src/coreclr/jit/gentree.cpp

```diff
@@ -7269,6 +7269,14 @@ bool GenTree::OperMayThrow(Compiler* comp)
         {
             return true;
         }
+
+#ifdef TARGET_XARCH
+        NamedIntrinsic intrinsicId = this->AsHWIntrinsic()->GetHWIntrinsicId();
+        if (intrinsicId == NI_Vector128_op_Division || intrinsicId == NI_Vector256_op_Division)
+        {
+            return true;
```

Comment:
This should probably assert that we're only encountering it for `varTypeIsInteger(AsHWIntrinsic()->GetSimdBaseType())`, just to avoid issues from potential future refactorings.

I'd expect we want to go ahead and include `Vector512_op_Division` here as well, even if it can't be encountered quite yet.

Reply:
Thanks, I've added the `assert` and included `Vector512_op_Division`.